### ISSUE TITLE
Connect to the wireless mouse,  the scrolling speed  is very slow when the scroll wheel slides

### ISSUE THREAD

**Issue by wanderingGuy** [flutter/flutter#67985]
Internal: b/241573125

The mobile phone is connected to the wireless ( or wired ) mouse. The scrolling speed of the list is very slow when the scroll wheel slides. I don't know whether the application layer can adjust it.

my Flutter version: 1.12.13

**Comment by darshankawar** [comment]
@wanderingGuy 
You seem to be on an obsolete version of Flutter. I recommend to upgrade to latest Stable version 1.22.1 and try again to see if it works as expected.
If problem persists after upgrading, please provide updated `flutter doctor -v` and a minimal complete reproducible code sample that shows the issue.
Thanks.

**Comment by wanderingGuy** [comment]
@darshankawar When I upgraded the flutter version to 1.22.1, the problem could still reproduce. 

Here is my test code：
```
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';

class TestListPage extends StatefulWidget {
  @override
  _TestListPageState createState() => _TestListPageState();
}

class _TestListPageState extends State<TestListPage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('List测试'),
      ),
      body: _buildCustomList(),
    );
  }

  //error
  _buildCustomList() {
    return ListView.builder(
      shrinkWrap: true,
      itemBuilder: (BuildContext context, int index) {
        return ListTile(title: Text('index $index'),);
      },
      itemCount: 100,
    );
  }
}
```

**My mouse wheel is spinning fast, but the list slides very slowly, compared to the native app.**

![list_gif](https://user-images.githubusercontent.com/17696836/95862637-c74e0780-0d95-11eb-93d9-cfe498b13301.gif)

flutter doctor -v

```
[✓] Flutter (Channel stable, 1.22.1, on Mac OS X 10.15.2 19C57, locale zh-Hans-CN)
    • Flutter version 1.22.1 at /Users/wandering/Documents/flutter_1.22.1
    • Framework revision f30b7f4db9 (5 days ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1
    • Pub download mirror https://dart-pub.byted.org
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/wandering/Documents/android-sdk-macosx
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/wandering/Documents/android-sdk-macosx
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

 
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.1

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.
```

**Comment by darshankawar** [comment]
I tried to replicate this issue by trying to connect my wireless mouse with iphone, but for some reason, it couldn't detect the mouse, hence I am unable to verify this issue. Keeping this open for further analysis.

**Comment by andymstone** [comment]
I see the same issue with Flutter 2.0.3 for any scrolling widgets on Android devices. 

If you use the scroll wheel on a wireless mouse connected to the phone the views scroll really slowly compared to non-Flutter apps.  I also tried using an app to emulate a wireless mouse on a second device with the same results - apps like Chrome scroll much more for similar movements of the 'wheel' (I used this to test: https://play.google.com/store/apps/details?id=io.appground.blek).

flutter doctor - v:

```
[√] Flutter (Channel stable, 2.0.3, on Microsoft Windows [Version
    10.0.19041.928], locale en-AU)
    • Flutter version 2.0.3 at c:\flutter\flutter
    • Framework revision 4d7946a68d (5 weeks ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\Andy\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b01)

[√] VS Code (version 1.55.1)
    • VS Code at C:\Users\Andy\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.20.0

[√] Connected device (3 available)
    • Nexus 5 (mobile) • 08322a2b2132209d • android-arm    • Android 6.0.1 (API
      23)
    • Chrome (web)     • chrome           • web-javascript • Google Chrome
      90.0.4430.72
    • Edge (web)       • edge             • web-javascript • Microsoft Edge
      90.0.818.42

```

**Comment by dragongesa** [comment]
same here. thats my problem since flutter beta

**Comment by zyrouge** [comment]
Bumping this up as this it is really noticeable making the overall app feel really slow.

**Comment by dragongesa** [comment]
i wonder if it is a feature for now 😄

**Comment by thegrxp** [comment]
Mouse scrolling is really slow on desktop as well

**Comment by thegrxp** [comment]
In the meantime, this does the job on desktop. Don't forget to `dispose()` your controller

```
_scrollController.addListener(() {
      ScrollDirection scrollDirection =
          _scrollController.position.userScrollDirection;
      if (scrollDirection != ScrollDirection.idle) {
        double scrollEnd = _scrollController.offset +
            (scrollDirection == ScrollDirection.reverse
                ? EXTRA_SCROLL_SPEED
                : -EXTRA_SCROLL_SPEED);
        scrollEnd = min(_scrollController.position.maxScrollExtent,
            max(_scrollController.position.minScrollExtent, scrollEnd));
        _scrollController.jumpTo(scrollEnd);
      }
    });
```

**Comment by selvam920** [comment]
any update, waiting for this fix

**Comment by ssaajjiinn** [comment]
Any update on this issue for Android?

**Comment by foreverpw** [comment]
Any update?

**Comment by lff5** [comment]
Hi. Any update?

**Comment by sky96111** [comment]
Seems fixed in flutter 3.7.0
```
Flutter 3.7.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b06b8b2710 (6 days ago) • 2023-01-23 16:55:55 -0800
Engine • revision b24591ed32
Tools • Dart 2.19.0 • DevTools 2.20.1
```

**Comment by lukehutch** [comment]
It's not fixed in Flutter 3.7.3, at least when using a mouse scrollwheel with the Android emulator on Linux. Each mousewheel event scrolls just a few pixels.

**Comment by sky96111** [comment]
OK. Previously this bug had a companion bug which made all flutter desktop apps scroll slow in the list.
Maybe 3.7.0 only fixes that companion bug.

**Comment by lukehutch** [comment]
Also it's not just a list view that has the problem. It's anything scrollable, e.g. `SingleChildScrollView`.

**Comment by selvam920** [comment]
waiting for this bug very long time

**Comment by gnprice** [comment]
Before adding a comment here, please read https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-bugs .

Comments like "any update" or "it's been a long time" do not help get the bug fixed. Instead they make the thread harder to read for anyone trying to understand and fix the bug.

**Comment by math1man** [comment]
Internal bug for huggsy: b/241573125

**Comment by wuzhengu** [comment]
Flutter (Channel stable, 3.7.6)也是
@Hixie

**Comment by davidmartos96** [comment]
This appears to be fixed on Flutter Beta 3.15.0-15.1.pre :tada: 

Commit in the engine that fixes it https://github.com/flutter/engine/commit/28b8bd5d5d913d6c684df6ac980edc47c4bfc52f

**Comment by gspencergoog** [comment]
This appears to be fixed.
